lib: libc: minimal: add check on returned value

gudnimg opened the pull request:
* Fix #32938 [Coverity CID :219508] Unchecked return value in `lib/libc/minimal/source/stdlib/malloc.c`
* Fix #32937 [Coverity CID :219508] Unchecked return value in `lib/libc/minimal/source/stdlib/malloc.c `

The Coverity complains about `sys_mutex_lock()` which returns `0` if
locked. I added also the same check on returned value for
`sys_mutex_unlock()` which returns `0` if unlocked.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

GunZi200 commented:
I'm wondering how we could track the returned value from `sys_mutex_lock()` and `sys_mutex_unlock()`. Could we use the `errno` variable?

Something like this: https://github.com/zephyrproject-rtos/zephyr/blob/ee79143ec80b1f9e83ab25d74a654517e214f303/lib/libc/minimal/source/stdlib/malloc.c#L51

gudnimg committed:
lib: libc: minimal: add check on returned value

Fix #32938 [Coverity CID :219508] "Unchecked return value in
lib/libc/minimal/source/stdlib/malloc.c"

The Coverity complains about sys_mutex_lock() which returns 0 if
locked. I added also the same check on returned value for
sys_mutex_unlock() which returns 0 if unlocked.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

GunZi200 commented:
Added a `(void)` cast to `sys_mutex_unlock()` in `free()` to make it clear we are not interested in the returned value. Hopefully that will be enough to supress a false alarm Coverity report.

GunZi200 commented:
@nashif friendly reminder :) do you know of anyone else to review this so we can proceed with the fix?
